### ISSUE TITLE
Pocket representation

### ISSUE THREAD

**Issue by jyaacoub** [jyaacoub/MutDTA#103]
# Pocket-only representation
To make sure we dont have to build entirely seperate datasets for the pocket representation, this implementation should just get index positions for our binding pocket and then just apply a mask to the original graph (similar to how it is done with the [`dropout_node`](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.dropout_node) function in `pytorch_geometric`).

**Task list:**
- [x] #109
- [x] #110
- [x] #123
- [ ] Train models and evaluate

## [KLIFS Database](https://klifs.net/details.php?structure_id=3834)
This is used by KBDNet to get the binding pockets for davis and kiba "pocket is 85 residues long".

The sequence given from KLIFS is not contiguous and only contains relevant pocket residues. 
- So to match it up with our structures we need to do a sequence alignment and then **get the index positions of those residues**.

After we get our list of index positions for the binding pocket AA we can modify our existing graph by applying a mask 
- This way we don't need to build entirely separate databases, and can just apply the mask before inference

## Getting pockets for Kiba
1. Using **UniProt ID** we can get the pocket from the KLIFS database with the `/kinase_ID` API.
	1. For example: https://klifs.net/api/kinase_ID?kinase_name=O00141&species=HUMAN returns: ![image](https://github.com/jyaacoub/MutDTA/assets/50300488/a9695c93-b739-428d-813f-49cac6ee3da3)

## Getting pockets for davis:
Same as for kiba, but we use the raw **Gene Name** code (need to remove any mutation or phosphorylation information): `ABL1(F317I)p` -> `ABL1`
	1. For example: https://klifs.net/api/kinase_ID?kinase_name=ABL1&species=HUMAN returns: 
![image](https://github.com/jyaacoub/MutDTA/assets/50300488/798e2caa-8ae7-4abe-a367-bbf4af5c289a)

However **for mutated genes**  we must be careful with the sequence alignment, and must follow the following procedure to get the right amino acid index positions:
1. Reverting the mutation
3. Perform alignment 
4. Extract index positions

then we just use these positions for our mask on the original (mutated) graph.

**Comment by jyaacoub** [comment]
# Building the pocket dataset
**Assumes that we have a normal dataset built already.**

## 1. get and mask for pockets with KLIFS
```python
# building pocket datasets:
from src.utils.pocket_alignment import pocket_dataset_full
import shutil
import os

data_dir = '/cluster/home/t122995uhn/projects/data/'
db_type = ['kiba', 'davis']
db_feat = ['nomsa_binary_original_binary', 'nomsa_aflow_original_binary', 
           'nomsa_binary_gvp_binary',      'nomsa_aflow_gvp_binary']

for t in db_type:
    for f in db_feat:
        print(f'\n---{t}-{f}---\n')
        dataset_dir= f"{data_dir}/DavisKibaDataset/{t}/{f}/full"
        save_dir   = f"{data_dir}/v131/DavisKibaDataset/{t}/{f}/full"
        
        pocket_dataset_full(
            dataset_dir= dataset_dir,
            pocket_dir = f"{data_dir}/{t}/",
            save_dir   = save_dir,
            skip_download=True
        )
        
``` 

## 2. resplit the database:
```python
import os
from src.data_prep.init_dataset import create_datasets
from src import cfg
import logging
cfg.logger.setLevel(logging.DEBUG)

dbs = [cfg.DATA_OPT.davis, cfg.DATA_OPT.kiba]
splits = ['davis', 'kiba']
splits = ['/cluster/home/t122995uhn/projects/MutDTA/splits/' + s for s in splits]
print(splits)

#%%
for split, db in zip(splits, dbs):
    print('\n',split, db)
    create_datasets(db, 
                feat_opt=cfg.PRO_FEAT_OPT.nomsa, 
                edge_opt=[cfg.PRO_EDGE_OPT.binary, cfg.PRO_EDGE_OPT.aflow],
                ligand_features=[cfg.LIG_FEAT_OPT.original, cfg.LIG_FEAT_OPT.gvp], 
                ligand_edges=cfg.LIG_EDGE_OPT.binary, overwrite=False,
                k_folds=5,
                test_prots_csv=f'{split}/test.csv',
                val_prots_csv=[f'{split}/val{i}.csv' for i in range(5)],)
                # data_root=os.path.abspath('../data/test/'))

```
